src/psyclone/psyGen.py

```diff
@@ -2470,11 +2367,10 @@ def local_reduction_name(self):
         '''Generate a local variable name that is unique for the current
         reduction argument name. This is used for thread-local
         reductions with reproducible reductions '''
-        var_name = self._reduction_arg.name
-        return self._name_space_manager.\
-            create_name(root_name="l_"+var_name,
-                        context="PSyVars",
-                        label=var_name)
+        tag = self._reduction_arg.name
+        name =  self.root.symbol_table.name_from_tag(tag, "l_" + tag)
```

Comment:
As we're planning to deprecate `name_from_tag`, can this be replaced here?

Reply:
It won't be easy to replace this instance, I added a TODO for #720 instead